services/secondary-storage/server/src/main/java/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java

```diff
@@ -2225,7 +2225,7 @@ private String deleteLocalFile(String fullPath) {
         return null;
     }
 
-    public String allowOutgoingOnPrivate(String destCidr) {
+    public synchronized String allowOutgoingOnPrivate(String destCidr) {
```

Comment:
```suggestion
    /**
    * allow *only one* setting of an outgoing destination at a time
    *
    * @destCidr the destination network that will be allowed for outgoing traffic.
    * @return any error message that might be helpful or <null> on success or when called anywhere but in the router VM.
    */
    public synchronized String allowOutgoingOnPrivate(String destCidr) {
```

Reply:
Could not be reproduced (the bug), so the solution as well could not be verified. The customer has confirmed (by testing in their env) that the "synchronized" seems to solve the issue on their end.